src/tests/ftest/osa/osa_offline_reintegration.py

```diff
@@ -24,111 +25,195 @@ def setUp(self):
+                self.pool.set_property("reclaim", "disabled")
             else:
-                output = self.dmg_command.system_stop(ranks=rank)
-                self.pool.wait_for_rebuild(True)
+                test_seq = self.ior_test_sequence[0]
+            if data:
+                self.run_ior_thread("Write", oclass, test_seq)
+                self.run_mdtest_thread()
+
+        # Exclude all the ranks
+        random_pool = random.randint(0, (num_pool-1))
+        for _ in range(0, self.loop_test_cnt):
+            for val, _ in enumerate(rank):
+                self.pool = pool[random_pool]
+                self.pool.display_pool_daos_space("Pool space: Beginning")
+                pver_begin = self.get_pool_version()
+                self.log.info("Pool Version at the beginning %s", pver_begin)
+                if server_boot is False:
+                    if (reint_during_rebuild is True and val == 0):
+                        # Exclude rank 5
+                        output = self.dmg_command.pool_exclude(self.pool.uuid,
+                                                               "5")
+                        self.log.info(output)
+                        self.is_rebuild_done(3)
+                        self.assert_on_rebuild_failure()
+                    if reint_during_aggregation is True:
+                        self.pool.set_property("reclaim", "time")
+                        time.sleep(90)
+                    output = self.dmg_command.pool_exclude(self.pool.uuid,
+                                                           rank[val])
+                else:
+                    output = self.dmg_command.system_stop(ranks=rank[val])
+                    self.log.info(output)
+                    self.is_rebuild_done(3)
+                    self.assert_on_rebuild_failure()
+                    output = self.dmg_command.system_start(ranks=rank[val])
+                # Just try to reintegrate rank 5
+                if (reint_during_rebuild is True and val == 2):
+                    # Exclude rank 5
+                    time.sleep(3)
+                    output = self.dmg_command.pool_reintegrate(self.pool.uuid,
+                                                               "5")
+                self.log.info(output)
+                self.is_rebuild_done(3)
+                self.assert_on_rebuild_failure()
+
+                pver_exclude = self.get_pool_version()
+                self.log.info("Pool Version after exclude %s", pver_exclude)
+                # Check pool version incremented after pool exclude
+                # pver_exclude should be greater than
+                # pver_begin + 3 (2 targets + exclude)
+                self.assertTrue(pver_exclude > (pver_begin + 3),
+                                "Pool Version Error: After exclude")
+
+            # Reintegrate the ranks which was excluded
+            for val, _ in enumerate(rank):
+                time.sleep(5)
```

Comment:
Please avoid sleep and use system query to find the expected status before/after any osa feature execution.

Reply:
Will be removed... Thanks.

Reply:
Done.